### PR TITLE
add generic torchmetric ppl logging on esm2

### PR DESCRIPTION
**Pull request by sichu2023**
## Summary
Logging perplexity through `torchmetric.text.Perplexity`

## Details
Pipeline parallel last stage is handled in `training_step` and `validation_step`. Generic torchmetric seems to handle tensor parallelism without any noticeable difference, where the logits are already provided in `training/validation_step` and no backward pass is needed.

However, valid_ppl does not seem to work properly. 

## Usage
```python
python sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/train_esm2.py --log_train_ppl --log_val_ppl
```

W B Chart 12_28_2024, 12_09_19 PM
W B Chart 12_28_2024, 12_09_25 PM
W B Chart 12_28_2024, 12_12_35 PM
W B Chart 12_28_2024, 12_12_40 PM

### CODE REVIEW

**Comment by sichu2023** [comment]
Moved to https://github.com/NVIDIA/bionemo-framework/pull/622
